### DISCUSSION TITLE
[Config Support]: Is it possible to run motion detection on GPU?

### DISCUSSION THREAD

**Discussion by ross-rosario** [blakeblackshear/frigate#10527] [Config Support] [answered]
### Describe the problem you are having

Currently GPU is responsible for video decoding and object detection, and it's ample enough to do more work if possible.

So I wonder, can motion detection as well be offloaded to the GPU, in order to free my CPU completely for other tasks?

Thank you.

### Version

latest

### Frigate config file

```yaml
na
```

### Relevant log output

```shell
na
```

### Operating system

Other Linux

### Install method

Docker Compose

### Coral version

Other

**Comment by NickM-27** [comment, marked as answer]
it is possible but will vary greatly from devices and the software used. Motion detection is comprised of a number of different parts and only some can be run on GPU. Making the GPU effort likely not worth it. We have other ideas for rewriting motion detection in the future to be more efficient like rewriting it in another language instead of python

**Reply by ross-rosario** [reply]
Thanks for the write up! Sorry to piggy back on that question but it's kind of the same theme.

What if my dahua camera does have motion detection? Can I, somehow, have Frigate rely on that instead of using the CPU?

**Edit**: obviously my purpose here is to make sure I free the CPU as much possible. Please let me know if there are other tweaks I could benefit from.

**Reply by NickM-27** [reply]
No, because cameras don't send enough information for frigate to know where to look in the frame and the latency would be too high for the real time object tracking that frigate does

**Reply by ross-rosario** [reply]
Makes sense. Thank you.

**Reply by titovskiy** [reply]
> No, because cameras don't send enough information for frigate to know where to look in the frame and the latency would be too high for the real time object tracking that frigate does

and what about to wait for onvif sensor to start check motion on frigate? camera send onvif to frigate => frigate starts detections
it will reduce CPU utilization if no motion or motion is not so

or maybe wait for occupancy same from camera detector
